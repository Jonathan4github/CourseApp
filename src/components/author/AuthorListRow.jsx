import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';


const propTypes = {
  author: PropTypes.shape().isRequired,
  handleDelete: PropTypes.func.isRequired
};

const AuthorListRow = ({ author, handleDelete }) => (
  <tr>
    <td><Link to={`/authors/${author.id}`}>{author.id}</Link></td>
    <td>{author.firstName}</td>
    <td>{author.lastName}</td>
    <td>
      <button
        type="button"
        onClick={() => handleDelete(author.id)}
        className="btn btn-outline-danger btn-sm">
        Delete
      </button>
    </td>
  </tr>
);

AuthorListRow.propTypes = propTypes;

export default AuthorListRow;
