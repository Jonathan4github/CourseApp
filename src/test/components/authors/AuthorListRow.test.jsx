/* eslint-disable */
import AuthorListRow from '../../../components/author/AuthorListRow';
import authorsMock from '../../__mocks__/authorMock';


describe('AuthorListRow component', () => {
  const setup = () => {
    const props = {
      author: authorsMock.author,
      handleDelete: jest.fn()
    };

    return {
      shallowWrapper: shallow(<AuthorListRow {...props} />),
      props
    };
  };


  it('should render without crashing', () => {
    const { shallowWrapper } = setup();

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('should invoke the handleDelete method', () => {
    const { shallowWrapper, props } = setup();

    shallowWrapper.find('button').simulate('click', {});

    expect(props.handleDelete).toHaveBeenCalledWith(props.author.id);
  });
});
