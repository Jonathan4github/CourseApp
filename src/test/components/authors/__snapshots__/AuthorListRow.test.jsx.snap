// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorListRow component should render without crashing 1`] = `
<tr>
  <td>
    <Link
      replace={false}
      to="/authors/cory-house"
    >
      cory-house
    </Link>
  </td>
  <td>
    Cory
  </td>
  <td>
    House
  </td>
  <td>
    <button
      className="btn btn-outline-danger btn-sm"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </td>
</tr>
`;
